File: MVVMIntro/app/src/main/java/com/example/mvvmintro/view_models/NoteViewModel.java
package com.example.mvvmintro.view_models;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.mvvmintro.entities.Note;
import com.example.mvvmintro.repositories.NoteRepository;

import java.util.List;

//  If you extend the viewmodel class and pass the context to it,
//  The application will outlive past the destroyed state hence causing
//  A memory leak
public class NoteViewModel extends AndroidViewModel {

    private NoteRepository repository;
    private LiveData<List<Note>> allNotes;

    public NoteViewModel(@NonNull Application application) {
        super(application);

        repository = new NoteRepository(application);
        allNotes = repository.getAllNotes();
    }

    public void insert(Note note) {
        repository.insert(note);
    }

    public void update(Note note) {
        repository.update(note);
    }

    public void delete(Note note) {
        repository.delete(note);
    }

    public void deleteAllNotes() {
        repository.deleteAllNotes();
    }

    public LiveData<List<Note>> getAllNotes() {
        return allNotes;
    }
}

File: CircularCheckBox/app/src/main/java/com/example/circularcheckbox/viewModels/NoteViewModel.kt
package com.example.circularcheckbox.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.example.circularcheckbox.entities.Notes
import com.example.circularcheckbox.repositories.NoteRepository

class NoteViewModel(application: Application): AndroidViewModel(application) {

    private val repository = NoteRepository(application)
    private val allTasks = repository.getAllTasks()

    fun modelInsertTask(note: Notes) {
        repository.insertTask(note)
    }

    fun modelUpdateTask(note: Notes) {
        repository.updateTask(note)
    }

    fun modelDeleteTask(note: Notes) {
        repository.deleteTask(note)
    }

    fun modelGetAllTasks(): LiveData<List<Notes>> {
        return allTasks
    }

    fun modelGetAllCheckedTasks(status: Boolean): LiveData<List<Notes>> {
        return repository.getAllCheckedTasks(status)
    }
}